integration-tests/smoke/keeper_test.go

```diff
@@ -171,35 +173,82 @@ func getKeeperSuite(
 					Expect(err).ShouldNot(HaveOccurred(), "Upkeep should get cancelled successfully")
 					err = networks.Default.WaitForEvents()
 					Expect(err).ShouldNot(HaveOccurred(), "Error waiting for cancel upkeep tx")
 
-					// Get existing performed count
+					// Obtain the amount of times the upkeep has been executed so far
 					existingCnt, err := consumer.Counter(context.Background())
 					Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
 					log.Info().Int64("Upkeep counter", existingCnt.Int64()).Msg("Upkeep cancelled")
 
-					// Expect count to be remain consistent
+					// Expect the counter to remain constant because the upkeep was cancelled, so it shouldn't increase anymore
 					Consistently(func(g Gomega) {
 						cnt, err := consumer.Counter(context.Background())
 						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's Counter shouldn't fail")
 						g.Expect(cnt.Int64()).Should(
 							Equal(existingCnt.Int64()),
-							"Expected consumer counter to to remain constant at %d, but got %d", existingCnt.Int64(), cnt.Int64(),
+							"Expected consumer counter to remain constant at %d, but got %d", existingCnt.Int64(), cnt.Int64(),
 						)
 					}, "1m", "1s").Should(Succeed())
+
+					// Register another 9 upkeeps and deploy the contracts associated to them
+					registry, consumers, upkeepIDs := actions.DeployKeeperContracts(
+						registryVersion,
+						registryConfig,
+						9,
+						uint32(2500000), //upkeepGasLimit
+						linkToken,
+						contractDeployer,
+						networks,
+					)
+
+					// Register the Keeper jobs responsible for the 9 newly registered upkeeps
+					actions.CreateKeeperJobs(chainlinkNodes, registry)
+					err = networks.Default.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
+
+					var testedUpkeeps = 0
+					var upkeepIndexesLeftToTest = []int{0, 1, 2, 3, 4, 5, 6, 7, 8}
+
+					// Go through all the 9 upkeeps and check if they are performing (increasing execution counter)
+					for testedUpkeeps < 9 {
+						var randomIndex = rand.Intn(len(upkeepIndexesLeftToTest))
+						var randomConsumerAndUpkeepID = upkeepIndexesLeftToTest[randomIndex]
+
+						log.Info().Msg("Watching upkeep indexed at " + strconv.Itoa(randomConsumerAndUpkeepID) +
+							" in the original list of 9 upkeeps perform")
+
+						consumer = consumers[randomConsumerAndUpkeepID]
+						upkeepID = upkeepIDs[randomConsumerAndUpkeepID]
+
+						// Check if the upkeep is being performed by verifying the counter is greater than 0
+						Eventually(func(g Gomega) {
+							cnt, err := consumer.Counter(context.Background())
+							g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+							g.Expect(cnt.Int64()).Should(BeNumerically(">", int64(0)),
+								"Expected consumer counter to be greater than 0, but got %d", cnt.Int64())
+							log.Info().Int64("Upkeep counter", cnt.Int64()).Msg("Upkeeps performed")
```

Comment:
We can simply loop over the upkeeps here in order. No need for randomization or multiple waits

Reply:
Ok sure, sorry if I over-complicated it :sweat_smile: